v2/dtos/requests/event_test.go

```diff
@@ -21,20 +22,116 @@ var testAddEvent = AddEventRequest{
 	},
-	DeviceName: TestDeviceName,
-	Origin:     TestOriginTime,
-	Readings:   nil,
+	Event: dtos.Event{
+		DeviceName: TestDeviceName,
+		Origin:     TestOriginTime,
+		Readings: []dtos.BaseReading{{
+			DeviceName: TestDeviceName,
+			Name:       TestDeviceResourceName,
+			Origin:     TestOriginTime,
+			ValueType:  TestReadingValueType,
+			SimpleReading: dtos.SimpleReading{
+				Value: TestReadingValue,
+			},
+		}},
+	},
 }
 
 func TestAddEventRequest_Validate(t *testing.T) {
 	valid := testAddEvent
 	noReID := testAddEvent
 	noReID.RequestID = ""
 	noDeviceName := testAddEvent
-	noDeviceName.DeviceName = ""
-	noDeviceOrigin := testAddEvent
-	noDeviceOrigin.DeviceName = ""
-	noDeviceOrigin.Origin = 0
+	noDeviceName.Event.DeviceName = ""
+	noOrigin := testAddEvent
+	noOrigin.Event.Origin = 0
+
+	noReading := testAddEvent
+	noReading.Event.Readings = nil
+
+	invalidReadingNoDevice := testAddEvent
+	invalidReadingNoDevice.Event.Readings = []dtos.BaseReading{{
+		DeviceName: "",
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		ValueType:  TestReadingValueType,
+		SimpleReading: dtos.SimpleReading{
+			Value: TestReadingValue,
+		},
+	}}
+	invalidReadingNoName := testAddEvent
+	invalidReadingNoName.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       "",
+		Origin:     TestOriginTime,
+		ValueType:  TestReadingValueType,
+		SimpleReading: dtos.SimpleReading{
+			Value: TestReadingValue,
+		},
+	}}
+	invalidReadingNoOrigin := testAddEvent
+	invalidReadingNoOrigin.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     0,
+		ValueType:  TestReadingValueType,
+		SimpleReading: dtos.SimpleReading{
+			Value: TestReadingValue,
+		},
+	}}
+	invalidReadingNoValueType := testAddEvent
+	invalidReadingNoValueType.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		ValueType:  "",
+		SimpleReading: dtos.SimpleReading{
+			Value: TestReadingValue,
+		},
+	}}
+
+	invalidSimpleReadingNoValue := testAddEvent
+	invalidSimpleReadingNoValue.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		ValueType:  TestReadingValueType,
+		SimpleReading: dtos.SimpleReading{
+			Value: "",
+		},
+	}}
+	invalidSRNoFloatEncoding := testAddEvent
+	invalidSRNoFloatEncoding.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		ValueType:  TestReadingFloatValueType,
+		SimpleReading: dtos.SimpleReading{
+			Value: TestReadingFloatValue,
+		},
+	}}
+
+	invalidBinaryReadingNoValue := testAddEvent
+	invalidBinaryReadingNoValue.Event.Readings = []dtos.BaseReading{{
+		DeviceName: TestDeviceName,
+		Name:       TestDeviceResourceName,
+		Origin:     TestOriginTime,
+		BinaryReading: dtos.BinaryReading{
```

Comment:
Need `ValueType: ValueTypeBinary` here and in all the Binary test cases. These should have failed validation on `ValueType` required.

Reply:
Updated the Binary test cases for adding `ValueType`. Thanks for catching that.